frontend/src/components/TabNavigation.tsx

```diff
@@ -0,0 +1,89 @@
+import styled from "@emotion/styled";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { Nav } from "react-bootstrap";
+import { IntlShape, useIntl } from "react-intl";
+import BreadcrumbItem from "./BreadcrumbItem";
+
+const StyledNav = styled(Nav)`
+  margin: 1rem 0;
+
+  .nav-item {
+  }
+
+  .nav-link {
+    &.active {
+    }
+  }
+`;
+
+export interface NavigationTab<T = undefined> {
+  url: string;
+  title: (intl: IntlShape, args: T) => string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-constraint
+const TabNavigation = <T extends unknown = undefined>({
+  tabs,
+  tabTitleArguments,
+  prefix,
+  breadcrumb,
+}: {
+  tabs: NavigationTab<T>[];
+
+  prefix?: string;
+  breadcrumb?: boolean;
+} & (T extends undefined
+  ? { tabTitleArguments?: T }
+  : { tabTitleArguments: T })) => {
+  const router = useRouter();
+  const intl = useIntl();
+
+  const navigationTabs = tabs.map((tab) => {
+    const url = prefix ? `${prefix}${tab.url}` : tab.url;
+
+    const urlWithoutLeadingAndTrailingSlash = url.replace(/(^\/|\/$)/g, "");
+    const currentPathWithoutLeadingAndTrailingSlash = router.asPath.replace(
+      /(^\/|\/$)/g,
+      "",
+    );
+
+    const isActive = currentPathWithoutLeadingAndTrailingSlash.startsWith(
+      urlWithoutLeadingAndTrailingSlash,
+    );
```

Comment:
This is an example of overly long names. (cf. Code Complete, p. 262 || p. 299 in the PDF)

You can be a lot more expressive by extracting a function or here simply the regex. E.g.

```suggestion
    // regex to remove both leading & trailing slashes
    const stripSlashRegex = /(^\/|\/$)/g;

    const strippedUrl = url.replace(stripSlashRegex, "");
    const strippedPath = router.asPath.replace(stripSlashRegex, "");

    const isActive = strippedPath.startsWith(strippedUrl);
```
(or `strippedCurrentPath` if you prefer)